libpod/oci_conmon.go

```diff
@@ -0,0 +1,7 @@
+package libpod
+
+const (
+	// cgroupSplitCurrent is the cgroup mode for reusing the current cgroup both
+	// for conmon and for the container payload.
+	cgroupSplitCurrent = "split-current"
```

Comment:
Just current

Reply:
I'd prefer `split` in the name, since the current cgroup is splitted in two (supervisor and container).  `current` sounds like the current cgroup is used without any further change.

Should I rename it to `split`?

Reply:
// cgroupSplitCurrent is the cgroup mode for reusing the current cgroup both
	// for conmon and for the container payload.
This description looks like you are using the same cgroup for conmon and the container?

I am thoroughly confused and `split` is not helping.  Could someone write down a table of the different modes we are looking at running cgroups

Can we get a modes table to descibe each one of these modes.

Reply:
I'll fix it.  Do you prefer if I fix the comment and explain better why it is `split-current` or just use `current`?

Reply:
<table>
<tr><th>Cgroups</th><th>Parent   Cgroup</th><th>conmon</th><th>container</th></tr>
<tr><td>disabled</td><td>any</td><td>unchanged</td><td>unchanged</td></tr>
<tr><td>enabled</td><td>blank</td><td>/machine.slice/libpod-conmon-CID.scope</td><td>/machine.slice/libpod-CID.scope</td></tr>
<tr><td>enabled</td><td>slice</td><td>/(Specified   Slice)/libpod-conmon-CID.scope</td><td>/(Specified   Slice)/libpod-CID.scope</td></tr>
<tr><td>no-conmon</td><td>default</td><td>unchanged</td><td>/machine.slice/libpod-CID.scope</td></tr>
<tr><td>no-conmon</td><td>slice</td><td>unchanged</td><td>/(Specified   Slice)/libpod-CID.scope</td></tr>
<tr><td>split-current</td><td>default</td><td>(current   cgroup)/supervisor</td><td>(current   cgroup)/container</td></tr>
<tr><td>split-current</td><td>slice</td><td>invalid   (slice/supervisor)</td><td>invalid   (slice/supervisor/slice/container)</td></tr>
</table>

Perhaps split-current should disallow cgroup-parent, or at the very least it should remove the "only a slice is valid" restriction because that validation assumes libpod will be manufacturing the scope name, which split-current doesn't do.

Split-current is nice because
1) it does what systemd-nspawn does
2) Systemd units get split nicely under the unit name, i.e.
  /system.slice/service.unit/supervisor -> conmon
  /system.slice/service.unit/container -> everything else

All processes show up nicely in `systemctl status`.  Slice can be changed in the unit if need be. (i.e. Slice=/machine.slice)

Ultimately using split-current w/ systemd units means we want to split the unit that systemd created, so I'm ok if `--cgroup-parent`` is disabled.

I get that you dislike hyphenated options.  This option already has `no-conmon`.  Current isn't correct, current is basically what "disabled" is, as is "reuse".  Split would make sense.

Reply:
Lets call it split.

Reply:
renamed and pushed